clients/build.gradle

```diff
@@ -104,22 +107,19 @@ configure(subprojects) {
     apply plugin: "swatch.java-conventions"
     apply plugin: "org.openapi.generator"
 
-    openApiValidate {
+    tasks.register("apiValidate", ValidateTask) {
         inputSpec = api_spec_path
     }
```

Comment:
I don't actually want to do this, because it means that we end up with extraneous tasks defined, see `./gradlew :clients:rhsm-client:tasks --all`. Additionally, gradle syntax and plugin implementation make this unclear, but actually, `openApiValidate` task is already declared via `tasks.register`. See [generator plugin source](https://github.com/OpenAPITools/openapi-generator/blob/d17316e8d98be750fd87af528174b6584578d394/modules/openapi-generator-gradle-plugin/src/main/kotlin/org/openapitools/generator/gradle/plugin/OpenApiGeneratorPlugin.kt#L82-L88).

`openApiValidate { /*...*/ }` defines [an extension instance](https://github.com/OpenAPITools/openapi-generator/blob/d17316e8d98be750fd87af528174b6584578d394/modules/openapi-generator-gradle-plugin/src/main/kotlin/org/openapitools/generator/gradle/plugin/extensions/OpenApiGeneratorValidateExtension.kt) (essentially just a pojo), that the plugin then consults when running `tasks.register`.

Additionally we lose some things when defining custom tasks (namely task `group` and `description`, and any defaults applied in the extension class or plugin class).

Reply:
Yeah, I know it's not ideal, but it leaves us in the position of having 2 different tasks that do the same thing.  Some projects have `openApiValidate` while the main project has `apiValidate` since we can't define `openApiValidate` twice.  Let me experiment and see if I can pull `tasks.named('openApiValidate')` and add a depends on to our two validation tasks for the 2 internal API specs.

Reply:
So I tried

```
tasks.named("openApiGenerate") {
    dependsOn(provider {
        tasks.findAll { task -> task.name.startsWith('buildApi') }
    })
}
```

In the main build file and Gradle dies saying "In plugin 'org.openapi.generator' type 'org.openapitools.generator.gradle.plugin.tasks.GenerateTask' property 'inputSpec' doesn't have a configured value."  So it looks like either we move everything to tasks named "generateApi" or we have an inconsistency where the generation task has a different name for the main project.  Maybe it's not a big deal since it's all dependent on `compilreJava` anyway.  I just call the generation tasks directly a lot so keeping consistency in naming what a goal here.

Reply:
@awood see #881